### DISCUSSION TITLE
Bug Report: Fill on vector layer expands lines

### DISCUSSION THREAD

**Discussion by miya-sans** [keiwando/toonsquid#321] [Q&A]
Device: iPad Pro (12.9 inch) (6th generation)
iOS Version: 16.4.1
Version: 1.9

https://github.com/keiwando/toonsquid/assets/42623210/65568546-c4a4-452d-a81f-9df62e4e5864

When using the fill tool on vector layers, other vector lines will expand once. This affects all settings of the fill tool and is only a feature of vector layers, not affecting fills on raster layers. This seems to affect line widths of all sizes, but is the most apparent with a thin line.

https://github.com/keiwando/toonsquid/assets/42623210/06ac1a4e-ed1a-478d-93b3-e6e1684a3d64

In addition, using the fill tool on a new layer, when there are only vector layers, will default to creating a raster layer. If you set the vector layer to fill reference, this will by default create new vector layers, this also doesn't remove the "Smooth" options denoted by the feather, which is usually not in vector fill. But I feel like there should be a Vector Mode option in the Fill tool options, the same way individual brushes have a Vector Mode, in case you using a all vector workflow.

**Comment by miya-sans** [comment]
Another instance where this bug can be reproduced:

1. Draw vector line
2. Fill
3. Select
4. Cut
5. Paste
6. There will be a small gap between where the line and fill is
7. Filling again will full in this gap

https://github.com/keiwando/toonsquid/assets/42623210/ea86308e-b8cd-4b55-aa3f-370d3801dce2

Attached close-up, as it can hard to see the gap in the video. 
![Open RPReplay_Final1685954015](https://github.com/keiwando/toonsquid/assets/42623210/347d0e8a-03e5-412f-aa0f-fa85cca7d001)

**Comment by keiwando** [comment]
The slight thickness increase is on purpose as a trade-off in order to avoid exactly the kind of hairline gaps you showed in the bottom video. The gaps happen where two vector shapes touch because of the way that the underlying vector rendering engine performs its antialiasing on the individual strokes. 
There are some approaches that I still need to evaluate as potential long-term solutions to avoid the gaps without any workarounds but those are quite large undertakings that don't have a clear timeline right now and might turn out to have other issues preventing them from being usable.

The smaller issue that the gaps appear again if you copy or cut from a vector layer in which you previously filled a region is a bug that I will fix with the next update.

**Comment by DWrightAnimation** [comment]
I don't know if this helps or not, but one work around is to have the lines and fill color as separate layers, just set the line layer as a fill reference, and the thickness of the line should be the same when you fill the layer below it with color.
